### PR TITLE
Remove unused dependency

### PR DESCRIPTION
**Pull request by davidfowl**
- The authorization stack doesn't depend on AspNetCore at all really

This brings up an interesting question, why is it tied to ASP.NET at all? Shouldn't the AuthorizationContext flow the HttpContext as well? 

/cc @HaoK @blowdart @Tratcher @lodejard

### CODE REVIEW

**Comment by blowdart** [comment]
Design decision. We really want to underscore you make authn decisions based on an identity, nothing more (if you can).

**Comment by leastprivilege** [comment]
If you need the context - you could inject it.

**Comment by blowdart** [comment]
Well you don't need to inject it, MVC will add it as part of it's authorization context, it's just not part of the base one.

**Comment by davidfowl** [comment]
> If you need the context - you could inject it.

Yea, but using the IHttpContextAccessor is kinda gross. It also implies that you could use this outside of http requests and the "current" http context would be null in this case. Also, it's not added by default in RC2, it needs to be added to the container manually (and it's not free either, there is a performance slight penalty when you using it).

> Well you don't need to inject it, MVC will add it as part of it's authorization context, it's just not part of the base one.

MVC won't be the only thing using this system, also MVC passes its `AuthorizationContext` as the resource.

Bottom line is that there's no real **AspNet** in this package name. Maybe it should be **Microsoft.Extensions.Authorization** (but then again who would want to use this in a console application :wink: )

**Comment by leastprivilege** [comment]
Yea, but using the IHttpContextAccessor is kinda gross.

How would I get to the context and the current User when using Mvc.Core (e.g. for web apis)

**Comment by davidfowl** [comment]
> How would I get to the context and the current User when using Mvc.Core (e.g. for web apis). Since there is no controller base class.

There's 2 options:

1. Full Poco

```C#
public class MyController
{
   [ActionContext]
   public ActionContext ActionContext { get; set; }
  
   public HttpContext => ActionContext.HttpContext;
}
```

1b (This might have changed in RC2)


```C#
public class MyController
{
   [ActionContext]
   public ControllerContext ControllerContext { get; set; }
  
   public HttpContext => ControllerContext.HttpContext;
}
```

2. Use the new lightweight controller base.

```C#
public class MyController : ControllerBase
{
   ...
}
```

**Comment by leastprivilege** [comment]
ok. thanks!

**Comment by leastprivilege** [comment]
> Bottom line is that there's no real AspNet in this package name. Maybe it should be Microsoft.Extensions.Authorization (but then again who would want to use this in a console application :wink: )

Well - aren't asp.net applications all console applications ;) ??

I don't really care in which namespace or assemly the API lives - but yes - strictly speaking this is asp.net independent.

**Comment by davidfowl** [comment]
> Well - aren't asp.net applications all console applications ;) ??

Yes, but not all console applications are ASP.NET applications.

> I don't really care in which namespace or assemly the API lives - but yes - strictly speaking this is asp.net independent.

It's not though.

**Comment by leastprivilege** [comment]
So why are we having this discussion?

**Comment by davidfowl** [comment]
The discussion is on the wrong thing, it should be on this issue https://github.com/aspnet/Security/issues/726 :wink:
